cxx-qt-gen/src/parser/qobject.rs

```diff
@@ -0,0 +1,479 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::syntax::{
+    attribute::{attribute_find_path, attribute_tokens_to_ident},
+    path::{path_angled_args_to_type_path, path_compare_str, path_to_single_ident},
+};
+use quote::format_ident;
+use std::collections::HashMap;
+use syn::{
+    spanned::Spanned, Error, Ident, ImplItem, Item, ItemEnum, ItemImpl, ItemStruct, Result, Type,
+    TypePath,
+};
+
+#[derive(Default)]
+pub struct ParserCxxQtData {
+    /// Map of the QObjects defined in the module that need generation
+    pub qobjects: HashMap<Ident, ParserQObject>,
+    /// Any use blocks end up in the CXX-Qt generated module
+    pub uses: Vec<Item>,
+}
+
+/// A representation of a QObject within a CXX-Qt ItemMod
+///
+/// This has initial splitting of Item's into relevant blocks, other phases will
+/// then mutate these Item's for generation purposes.
+#[derive(Default)]
+pub struct ParserQObject {
+    /// This is the Data struct that currently stores the properties for the QObject
+    ///
+    /// In the future this will be removed
+    pub data_struct: Option<ItemStruct>,
+    /// This is the RustObj struct that stores the invokables for the QObject
+    pub rust_struct: Option<ItemStruct>,
+    /// This is the Signals enum that defines the Q_SIGNALS for the QObject
+    pub signal_enum: Option<ItemEnum>,
+    /// This is the list of methods that need to be implemented on the C++ object in Rust
+    ///
+    /// Ones marked as invokable will need to be exposed to C++ as Q_INVOKABLE
```

Comment:
How are the invokable methods marked as such in this struct?

Reply:
This hasn't happened yet but eventually we'll have like `ParsedInvokable` that has a bool or something to say `is_invokable` or be part of an enum etc. I'll change the comment to be a TODO.